parametric/test_128_bit_traceids.py

```diff
@@ -28,9 +28,151 @@ def test_datadog_128_bit_propagation(test_agent, test_library):
+                ["x-datadog-tags", "_dd.p.tid=1234567890abcde"],
+            ],
+        )
+    span = get_span(test_agent)
+    trace_id = span.get("trace_id")
+    dd_p_tid = span["meta"].get("_dd.p.tid")
+
+    assert trace_id == 1234567890123456789
+    assert int(headers["x-datadog-trace-id"], 10) == trace_id
+    assert dd_p_tid is None
+
+
+@pytest.mark.skip_library("dotnet", "not implemented")
+@pytest.mark.skip_library("golang", "not implemented")
+@pytest.mark.skip_library("java", "not implemented")
+@pytest.mark.skip_library("nodejs", "not implemented")
+@pytest.mark.skip_library("php", "Issue: Traces not available from test agent")
+@pytest.mark.skip_library("python", "not implemented")
+@pytest.mark.skip_library("python_http", "not implemented")
+@pytest.mark.skip_library("ruby", "not implemented")
+@pytest.mark.parametrize(
+    "library_env", [{"DD_TRACE_PROPAGATION_STYLE": "Datadog", "DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED": "false",}],
+)
+def test_datadog_128_bit_propagation_tid_chars(test_agent, test_library):
+    """ Ensure that incoming tids with bad characters are discarded.
+    """
+    with test_library:
+        headers = make_single_request_and_get_inject_headers(
+            test_library,
+            [
+                ["x-datadog-trace-id", "1234567890123456789"],
+                ["x-datadog-parent-id", "987654321"],
+                ["x-datadog-tags", "_dd.p.tid=1234567890abcdeX"],
+            ],
+        )
+    span = get_span(test_agent)
+    trace_id = span.get("trace_id")
+    dd_p_tid = span["meta"].get("_dd.p.tid")
+
+    assert trace_id == 1234567890123456789
+    assert int(headers["x-datadog-trace-id"], 10) == trace_id
+    assert dd_p_tid is None
+
+
+@pytest.mark.skip_library("dotnet", "not implemented")
+@pytest.mark.skip_library("golang", "not implemented")
+@pytest.mark.skip_library("java", "not implemented")
+@pytest.mark.skip_library("nodejs", "not implemented")
+@pytest.mark.skip_library("php", "Issue: Traces not available from test agent")
+@pytest.mark.skip_library("python", "not implemented")
+@pytest.mark.skip_library("python_http", "not implemented")
+@pytest.mark.skip_library("ruby", "not implemented")
+@pytest.mark.parametrize(
+    "library_env", [{"DD_TRACE_PROPAGATION_STYLE": "Datadog", "DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED": "false",}],
+)
+def test_datadog_128_bit_propagation_tid_malformed_optional_tag(test_agent, test_library):
+    """ Ensure that if incoming tids are malformed and the error is tagged, the tag is set to the expected value.
+    """
+    with test_library:
+        make_single_request_and_get_inject_headers(
+            test_library,
+            [
+                ["x-datadog-trace-id", "1234567890123456789"],
+                ["x-datadog-parent-id", "987654321"],
+                ["x-datadog-tags", "_dd.p.tid=XXXX"],
+            ],
+        )
+    assert get_span(test_agent)["meta"].get("_dd.propagation_error") == "malformed_tid XXXX"
```

Comment:
I think there are two issue with this assertion:
- Adding this tag is optional. This test should allow for no `_dd.propagation_error` tag.
- The `_dd.propagation_error` tag can be used for other problems in `x-datadog-tags`. Valid values are:
  - `extract_max_size`
  - `inject_max_size`
  - `disabled`
  - `encoding_error`
  - `decoding_error`

[source](https://docs.google.com/document/d/1zeO6LGnvxk5XweObHAwJbK3SfK23z7jQzp7ozWJTa2A/edit#heading=h.j993k37q7jo4) (internal)

Reply:
Same for other places where the tag is checked in this PR.

Reply:
I talked to Dino and he explained the idea is that only languages that support this will enable the test and other can skip it as "not supported." Sounds good to me, probably a better way of tracking which languages support an optional feature.

Thanks, Dino.